components/automate-cli/pkg/verifyserver/services/externalopensearchservice/externalopensearch.go

```diff
@@ -0,0 +1,106 @@
+package externalopensearchservice
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+)
+
+type IExternalOpensearchService interface {
+	GetExternalOpensearchDetails(models.ExternalOS, int) models.ExternalOpensearchResponse
+}
+
+type ExternalOpensearchService struct {
+	log     logger.Logger
+	timeout time.Duration
+}
+
+func NewExternalOpensearchService(log logger.Logger, timeout time.Duration) IExternalOpensearchService {
+	return &ExternalOpensearchService{
+		log:     log,
+		timeout: timeout,
+	}
+}
+
+func (eos *ExternalOpensearchService) checkReachability(reqBody models.ExternalOS, port int) models.ExternalOpensearchCheck {
+	eos.log.Debug("Checking External Opensearch Reachability")
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(reqBody.OSCert))
+	// Created HTTP client with a root_ca and timeout
+	client := &http.Client{
+		Timeout: eos.timeout * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			},
+		},
+	}
+
+	// Create a new request with basic authentication
+	url := fmt.Sprintf("https://%s:%d", reqBody.OSDomainURL, port)
+	eos.log.Debug("URL: ", url)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		eos.log.Debug("Failed to create request:", err)
+		return createExternalOpensearchCheck(false, constants.EXTERNAL_OPENSEARCH_FAILED_TITLE, constants.STATUS_FAIL, "", constants.EXTERNAL_OPENSEARCH_ERROR_MSG, constants.EXTERNAL_OPENSEARCH_RESOLUTION_MSG, fmt.Sprintf("Failed to create request: %v", err))
```

Comment:
This function is called multiple times with almost same arguments, create. function for the request if you received the error, append the error and the call this